### PR TITLE
Arithmetic opcode

### PR DESCRIPTION
**Pull request by matias-gonz**

### CODE REVIEW

File: gnark_backend_ffi/backend/plonk/structs/arithmetic_opcode_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		log.Fatal(err)
+	}
+
+	assert.NoError(t, err)
+	assert.Equal(t, backend.UncheckedDeserializeMulTerms(mulTerms), r.MulTerms)
+	assert.Equal(t, backend.UncheckedDeserializeAddTerms(addTerms), r.AddTerms)
+	assert.Equal(t, nonEncodedConstantTerm, r.qM)
+}
+
+func TestRawGatesTermUnmarshalJSON(t *testing.T) {
```
Comment (ilitteri) [inline gnark_backend_ffi/backend/plonk/structs/arithmetic_opcode_test.go:38]:
```suggestion
func TestArithmeticOpcodesTermUnmarshalJSON(t *testing.T) {
```

**Review by ilitteri — CHANGES_REQUESTED** [review]
Good work! Left some comments.

File: gnark_backend_ffi/backend/plonk/structs/arithmetic_opcode.go [hunk 2]
```diff
@@ -1,13 +1,76 @@
...
 	AddTerms []backend.AddTerm
 	qM       fr_bn254.Element
 }
+
+func (g *ArithmeticOpcode) UnmarshalJSON(data []byte) error {
+	var gateMap map[string]interface{}
+	err := json.Unmarshal(data, &gateMap)
+	if err != nil {
+		log.Print(err)
+		return err
```
[the remark this answers is not in the document]
Comment (matias-gonz) [reply gnark_backend_ffi/backend/plonk/structs/arithmetic_opcode.go:22]: It doesn't have extra tabs, GitHub just shows a very long space

[the remark this answers is not in the document]
Comment (ilitteri) [reply gnark_backend_ffi/backend/plonk/structs/arithmetic_opcode.go:22]: Lol ok

**Pull request merged**